nextdns/resource_nextdns_profile.go

```diff
@@ -43,6 +44,26 @@ func resourceNextDNSProfileCreate(ctx context.Context, d *schema.ResourceData, m
 	return resourceNextDNSProfileRead(ctx, d, meta)
 }
 
+func formatProfileIPv6Suffix(profileID []byte) []byte {
+	re := regexp.MustCompile(`(?P<quartet>[a-f0-9]{4})`)
+
+	reverseProfileID := make([]byte, len(profileID))
+	for i := len(profileID); i > 0; i-- {
+		reverseProfileID[len(profileID)-i] = profileID[i-1]
+	}
+
+	reverseFormatted := re.ReplaceAll(reverseProfileID, []byte("$quartet:"))
+	formatted := make([]byte, len(reverseFormatted))
+	for i := len(reverseFormatted); i > 0; i-- {
+		formatted[len(reverseFormatted)-i] = reverseFormatted[i-1]
+	}
+
+	if formatted[0] == ':' {
+		return formatted[1:]
+	}
```

Comment:
This is annoying, but I couldn't see how to avoid it as Go's `regexp` doesn't support negative look-ahead - i.e. no way to say "each quartet, unless it's at the end" without including the next (potentially _not_ the end) character in the match, and thus removing it from the next quartet.